### PR TITLE
#13352: Remove usage of legacy script in workflows

### LINKED ISSUE #13352: Retire legacy script `build_with_profiler_opt.sh`
We can now retire this from the repo:

https://github.com/tenstorrent/tt-metal/blob/main/scripts/build_scripts/build_with_profiler_opt.sh

It is only used in one workflow, and its usage can be replaced by either `build_metal.sh` or with direct CMake usage.

cc: @mo-tenstorrent

### PR DESCRIPTION
**Pull request by blozano-tt**
### Ticket
Closes #13352 

### Problem description
This old script for building with profiler keeps causing problems.
It is not tested in post-commit.
It is redundant now, since changes to build_metal.sh

### What's changed
Use build_metal.sh in CI workflow.

### Checklist
https://github.com/tenstorrent/tt-metal/actions/runs/11372491819
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/tutorials/005.ipynb [hunk 1]
```diff
@@ -612,7 +612,7 @@
     "cd $TT_METAL_HOME\n",
     "unset SILENT\n",
     "make nuke\n",
```
Comment (mo-tenstorrent) [inline ttnn/tutorials/005.ipynb:614]: Out side of the scope but is this nuke still relavant?

**Review by mo-tenstorrent — APPROVED** [review]

File: ttnn/tutorials/005.ipynb (hunk 1 above)
Comment (blozano-tt) [reply ttnn/tutorials/005.ipynb:614]: no idea, since its out of scope, someone who cares about that ipynb can fix it.

Comment (tt-rkim) [reply ttnn/tutorials/005.ipynb:614]: Haha this is real documentation too, oh boy

**Comment by ttmchiou** [comment]
please make sure to tell devs about this change.
There are probably more than a few who will be surprised that scripts/build_scripts/build_with_profiler_opt.sh is gone

**Comment by blozano-tt** [comment]
> please make sure to tell devs about this change. There are probably more than a few who will be surprised that scripts/build_scripts/build_with_profiler_opt.sh is gone

Done

**Pull request merged**